server/src/main/java/io/crate/rest/action/SqlHttpHandler.java

```diff
@@ -294,12 +295,19 @@ private CompletableFuture<XContentBuilder> executeBulkRequest(Session session,
     }
 
     Role userFromAuthHeader(@Nullable String authHeaderValue) {
-        String username = Headers.extractCredentialsFromHttpBasicAuthHeader(authHeaderValue).v1();
-        // Fallback to trusted user from configuration
-        if (username == null || username.isEmpty()) {
-            username = AuthSettings.AUTH_TRUST_HTTP_DEFAULT_HEADER.get(settings);
+        Credentials credentials = null;
+        try {
```

Comment:
why not use `try (Credentials credentials = Headers.....)` and remove the finally?

Reply:
Good point, updated